src/mlpack/bindings/go/CMakeLists.txt

```diff
@@ -187,33 +187,33 @@ if (BUILD_GO_BINDINGS)
 
   # Create ${name}.h.
   add_custom_command(OUTPUT
-      ${CMAKE_BINARY_DIR}/src/mlpack/bindings/go/src/mlpack.org/v1/mlpack/capi/${name}.h
+      ${CMAKE_CURRENT_BINARY_DIR}/src/mlpack/bindings/go/src/mlpack.org/v1/mlpack/capi/${name}.h
       COMMAND ${CMAKE_COMMAND}
           -DPROGRAM_NAME=${name}
           -DPROGRAM_MAIN_FILE=${CMAKE_CURRENT_SOURCE_DIR}/${directory}/${name}_main.cpp
-          -DSOURCE_DIR=${CMAKE_SOURCE_DIR}
-          -DGO_IN=${CMAKE_SOURCE_DIR}/src/mlpack/bindings/go/go_method.h.in
-          -DGO_OUT=${CMAKE_BINARY_DIR}/src/mlpack/bindings/go/src/mlpack.org/v1/mlpack/capi/${name}.h
-          -P ${CMAKE_SOURCE_DIR}/CMake/go/ConfigureGoHCPP.cmake
-      DEPENDS ${CMAKE_SOURCE_DIR}/src/mlpack/bindings/go/go_method.h.in
-              ${CMAKE_SOURCE_DIR}/CMake/go/ConfigureGoHCPP.cmake)
+          -DSOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}
```

Comment:
This needs to be the root of the mlpack repository... I guess you could do it relatively? `${CMAKE_CURRENT_SOURCE_DIR}/../../../../`?

Alternately you could set an `MLPACK_ROOT` variable or something like this at the top level?  Your call.